docs/content/development/extensions-contrib/moving-average-query.md

```diff
@@ -0,0 +1,357 @@
+|aggregations|Aggregations forms the input to Averagers; See [Aggregations](../../querying/aggregations.html)|yes|
+|postAggregations|Supports only aggregations as input; See [Post Aggregations](../../querying/post-aggregations.html)|no|
+|intervals|A JSON Object representing ISO-8601 Intervals. This defines the time ranges to run the query over.|yes|
+|context|An additional JSON Object which can be used to specify certain flags.|no|
+|averagers|Defines the moving average function; See [Aggregations](../../querying/aggregations.html)|yes|
+|postAveragers|Support input of both averagers and aggregations; Syntax is identical to postAggregations (See [Post Aggregations](../../querying/post-aggregations.html))|no|
+
+##Averagers
+
+Averagers are used to define the Moving-Average function. Averagers are not limited to an average - they can also provide other types of window functions such as MAX()/MIN().
+
+### Properties
+
+These are properties which are common to all Averagers:
+
+|property|description|required?|
+|--------|-----------|---------|
+|type|Averager type; See [Averager types](#averager-types)|yes|
+|name|Averager name|yes|
+|fieldName|Input name (An aggregation name)|yes|
+|buckets|Number of lookback buckets (time periods), including current one. Must be >0|yes|
+|cycleSize|Cycle size; Used to calculate day-of-week option; See [Cycle size (Day of Week)](#cycle-size-day-of-week)|no, defaults to 1|
+
+
+###Averager types:
+
+* [Standard averagers](#standard-averagers):
+  * doubleMean
+  * doubleMeanNoNulls
+  * doubleMax
+  * doubleMin
+  * longMean
+  * longMeanNoNulls
+  * longMax
+  * longMin
+* [Sketch averagers](#sketch-averagers):
+  * sketchUnion
+
+#### Standard averagers
+
+These averagers offer four functions:
+* Mean (Average)
+* MeanNoNulls (Ignores empty buckets).
+* Max
+* Min
+
+**Ignoring nulls**:
+Using a MeanNoNulls averager is useful when the interval starts at the dataset beginning time. 
+In that case, the first records will ignore missing buckets and average won't be artificially low.
+However, this also means that empty days in a sparse dataset will also be ignored.
+
+Example of usage:
+```json
+{ "type" : "doubleMean", "name" : <output_name>, "fieldName": <input_name> }
+```
+
+#### Sketch averagers
+
+Sketch averager are meant to perform merge operations on [DataSketches](../extensions-core/datasketches-extension.html) (When using a Sketch averager, please include the [DataSketches](../extensions-core/datasketches-extension.html) extension as well).
+
+Extra properties for Sketch averagers:
+
+|property|description|required?|
+|--------|-----------|---------|
+|size|Sketch size; See [DataSketches aggregator](../extensions-core/datasketches-aggregators.html)|no, defaults to 4096|
+
+Available functions: 
+* sketchUnion
+
+Example of usage:
+```json
+{ "type" : "sketchUnion", "name" : <output_name>, "fieldName": <input_name> }
+```
+
+### Cycle size (Day of Week)
```

Comment:
Is this a popular term for moving average?

Reply:
I couldn't find an exact similar term, but it can be achieved through advanced use of SQL.
I guess I need to explain this better in the documentation.
This is a pretty specific use case we have internally, where we want to further reduce the fluctuation of moving average by averaging a specific day-of-week. I.e. average over the last 4 Mondays.
This query supports it with `buckets`=28 AND `cycleSize=4` AND "first day of interval"==Monday.
I can also keep this undocumented and just explain via java doc.

Reply:
Oh, I think this is a cool feature. Please keep this documented. I could understand what it is from the description and the example, but the term was just not intuitive to me. Not sure what's a better name though.

Reply:
It's meant to be derived from [cyclic groups](https://en.wikipedia.org/wiki/Cyclic_group). Actually, it was called `skip` before that, which is even worse :)
I'm not sure what would be a better name but I hope for a revelation along the road.